src/lib.rs

```diff
@@ -230,8 +274,10 @@ impl Default for MemoryLayout {
             base_address: 0x2001_0000,
             tx_area_size: 0x2000,
             rx_area_size: 0x2000,
-            // Trace is not implemented yet
+            #[cfg(not(feature = "modem-trace"))]
             trace_area_size: 0,
+            #[cfg(feature = "modem-trace")]
+            trace_area_size: 0x2000,
```

Comment:
```suggestion
            trace_area_size: if cfg!(feature = "modem-trace") { 0x2000 } else { 0 },
```

This is better because all code is evaluated by the compiler and RA.
Might not pass rustfmt like this though :P